crates/core/tedge_api/src/entity_store.rs

```diff
@@ -327,10 +329,10 @@ impl EntityStore {
                 let topic = Topic::new(&format!("{MQTT_ROOT}/{parent_device_id}")).unwrap();
                 let device_register_message =
                     Message::new(&topic, device_register_payload).with_retain();
+                let device_register_message =
+                    EntityRegistrationMessage::try_from(&device_register_message).unwrap();
```

Comment:
I find weird the way this `EntityRegistrationMessage` is built, i.e. by parsing a message payload.

Why not directly building an `EntityRegistrationMessage`? This would avoid to:
- `unwrap()`
- Use an hard-coded `MQTT_ROOT`